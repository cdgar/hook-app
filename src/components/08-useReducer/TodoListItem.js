import React from "react";
import PropTypes from "prop-types";

const TodoListItem = ({ index, todo, handleToggleTodo, handleDeleteTodo }) => {
  return (
    <li className="list-group-item"  key={todo.id}>
      <p
        onClick={() => handleToggleTodo(todo.id)}
        className={`${todo.done && "complete"}`}
      >
        {index + 1}. {todo.desc}
      </p>
      <button
        onClick={() => handleDeleteTodo(todo.id)}
        className="btn btn-danger"
      >
        Delete
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  index: PropTypes.number.isRequired,
  todo: PropTypes.object.isRequired,
  handleToggleTodo: PropTypes.func.isRequired,
  handleDeleteTodo: PropTypes.func.isRequired,
};

export default TodoListItem;
