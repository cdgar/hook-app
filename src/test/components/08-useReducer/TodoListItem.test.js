import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";

describe('TodoListItem.js', () => {

    const handleDeleteTodo = jest.fn();
    const handleToggleTodo = jest.fn();
    const index = 0;
    const todo = demoTodos[0];
    let wrapper = shallow(
        <TodoListItem 
            index={index} 
            todo={todo} 
            handleToggleTodo={handleToggleTodo} 
            handleDeleteTodo={handleDeleteTodo}  
        />
    );

    test('should show TodoListItem', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should call handleDeleteTodo', () => {
        wrapper.find('button').simulate('click');
        expect(handleDeleteTodo).toHaveBeenCalledWith(todo.id);
    })

    test('should call handleToggleTodo', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggleTodo).toHaveBeenCalledWith(todo.id);
    })

    test('should check text', () => {
        expect(wrapper.find('p').text().trim()).toBe(`${index + 1}. ${todo.desc}`);
    })

    test('should check class', () => {
        todo.done = true
        wrapper = shallow(
            <TodoListItem 
                index={index} 
                todo={todo} 
                handleToggleTodo={handleToggleTodo} 
                handleDeleteTodo={handleDeleteTodo}  
            />
        );
        expect(wrapper.find('.complete').exists()).toBe(todo.done);
    })
    
});