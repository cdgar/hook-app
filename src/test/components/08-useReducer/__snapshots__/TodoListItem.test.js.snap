// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem.js should show TodoListItem 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    todo1
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
